### DISCUSSION TITLE
blog/how-to-use-roam-research

### DISCUSSION THREAD

**Discussion by giscus** [Pathsis/2750-words#30] [Announcements]
# blog/how-to-use-roam-research

哲学片段与学术之路

https://pathos.page/blog/how-to-use-roam-research

**Comment by qlibp** [comment]
好奇latex写作在roam里具体怎么实践

**Reply by Pathsis** [reply]
我现在已经不建议在 Roam 中写作了。因为，它现在有致命性的中文编辑问题（随机性地键入字母而不是汉字），而且始终没有见到修复。毕竟开发者是外国人，与用户的交流又稀烂。我已经准备全面放弃 Roam。

**Reply by qlibp** [reply]
确实有这个现象，我还以为是什么操作问题…不过兜兜转转了一圈，还是觉得roam比较适合我。
不过以你的文字构建的能力，感觉不太需要roam这种松散的记录方式，就可以用线性的文字构造出丰富的内容结构，用obsidian可能就挺好了

**Reply by Pathsis** [reply]
我对 Roam 始终爱不释手，主要是离不开它的大纲编辑器。我自己的 Roam 实践倒没那么松散，反倒比以往任何笔记软件都更有条理。和您感觉一样，每次打算切换到 Obsidian，始终找不到 Roam 那种秩序感。也考虑过 Logseq，但我实在受不了那完全不懂开发管理的团队日更一堆bug，以及一堆糟糕的细节。

**Reply by qlibp** [reply]
自从roam更新了diagram以后，我对它就没有更多的需求，所以我就又从obsidian转了回来。

近两年热度下来之后，我不确定roam团队是在克制他们的设计还是单纯因为开发人员节奏比较慢，大更新比较少。

但是他们的设计和特性让我更容易进入思考的状态，这是一种很“神奇”的体验。我用obsidian很难进入这种状态。

“花里胡哨”的特性少的情况下，少了“电子玩具”属性，更多让我专注于思考。刚开始的时候热衷于各种js/css做diy，后面就发现原生的特性足够我们做流畅的思考了，就不再折腾了。

**Reply by Pathsis** [reply]
那看来您对 Roam 是真爱。

Roam 的开发团队，怎么说呢，它的创始人给我一种玩世不恭或者说自由自在的美国人的感觉（据说他现在有三个妻子），就是那种在短时间爆发创造力，挣了钱就去玩乐了的人。这与比如说 Notion 这样的开发团队完全不同。

我感觉 Roam 的开发绝对是停滞了，而且不太可能重新活跃更新。等待它重新活跃更新，可能就像等待一个沉溺游戏的人，哪一天突然不想打游戏了，又重新回来爆发创造力。

不过我对 Roam 的功能完全满足了，如果有的话，可能就是  AI 吧。现在的问题还是，几乎难以正常使用的编辑器。

**Reply by qlibp** [reply]
看来我还是信息过于闭塞了，原来创始人还有这种八卦…

现在唯一的怨念就是数据不能在国内同步，速度时常让我难受，比输入偶尔出bug更难接受。

幻想着有一款自己的手机端可以接到自己的local-graph server，实现自己的笔记自由。

**Comment by ottokangjian** [comment]
非常有启发的文章，非常有启发的博客。感谢作者。也许互联网上“效率成瘾”的人们缺少的就是作者这样的抽象概括能力。现在的Obsidian已经成了新的roamcult，markdown成了政治正确。可殊不知，无论cult不cult，正确不正确，我们都不应该以手段去混淆目的。所有的工具都是手段，唯有目的才是目的。

最近也在思考这些问题。现在的做法是，用Obsidian管理半衰期短、复用性差的流水账、日志、扫描件等信息，用大纲笔记负责阅读、写作和深度思考的笔记。无奈，双链大纲软件个个都有硬伤，从rr到lsq，短板、缺陷一个赛一个明显。

读完这篇文章，大概明白作者的做法。尝试复述一下：highlight、批注属于“A阶段笔记”；review A阶段笔记的笔记属于“B阶段笔记”，根据“A+B”或其他来源的信息形成输出的作品属于C阶段（文章）、D阶段（项目/书籍）的成果。这里的A/B阶段，和涂子沛在第二大脑中所提到的dailynote到概念的笔记法高度一致。这个过程，也只有双链大纲软件能胜任。

在以上的前提下，2025年的Roam Research和LogSeq，如何做选择取舍？希望也能听听作者的建议。谢谢！

**Reply by qlibp** [reply]
Roam Research和LogSeq基本一致吧，LogSeq移动端更丝滑方便记录，但由于是开源的，所以质量把控上和roam比起来还是有点差距，但是并不大。ui风格我还是更偏好roam。

**Reply by Pathsis** [reply]
1. 我数次尝试换到 Obsidian，但每次都感觉不舒服：一个恰恰是纯文本编写带来的凌乱感；一个是其窗口布局不是 Roam 那样的左右主次布局带来的凌乱感。
2. 大纲编辑器确实有很多局限，也让很多人难以适应。但一旦适应，真的离不开。我日常写文章都是用大纲编辑器，用单纯的 markdown 甚至找不到感觉。不知道您是觉得 Roam 或 Logseq 这样的笔记软件的主要短板或缺陷是什么呢？
3. 您说的很是，我文章展示的这种笔记法，确实只有主次窗口布局和双链笔记（特别是前者）软件才能胜任。这是我迟迟难以离开 Roam 的原因。相比之下，WorkFlowy 迟迟不提供右侧边栏布局，这使得笔记对照成了很麻烦的事情。
4. 其实取舍上，我在《[笔记软件的选择问题](https://pathos.page/blog/notetaking-app-choice/)》一文中也做了很多维度的分析，首先看你 markdown（或纯文本）导向还是大纲编辑器导向。在大纲编辑器内，我会选择 Roam Research。Logseq 真的一言难尽，它仍然是 Roam Research 的拙劣模仿者，而且看起来不会改变了（因为作者模仿但又想不太一样，但自己有没有很好的想法）。
5. 当然，Logseq 的主要问题还是不会开发管理。现在都几年了？它还在搓 alpha 版。双链的风都吹过多久了，就算出来也会立即死掉，谁还在关心它？还能有多少用户？看看迟到两年的 Heptabase 的开发状况，那才是会开发软件的！
6. Roam Research 的主要问题有两个，一个是生命周期，一个是迁移数据。今年应该是第一批 beliver 到期的年份，相信 Roam Research 的用户会断崖式流失，恐怕 Roam 的开发会更加难以为继。但我相信再撑个五年应该没问题。假如 Roam 死了，它可能会开源，或者被收购持续维护，我相信它不会完全消失，因为它真的很优秀。我曾和其开发者交流过这个事情，开发者问了 Conor ，他说到时候可能会考虑这件事。当然，不确定性仍然存在。到时候，仍然可以设法几乎无损导入 Logseq。
7. 我目前摇摆于 WorkFlowy 和 Roam Research。双链很好！但我觉得双链还是可以舍弃的东西（你真的利用你的双链了吗）。但我仍然觉得 WorkFlowy 的窗口布局不如 Roam。
8. 既然选择 Roam，就要做好一个准备，哪一天真的完全不能用了，就导出到 Obsidian，起码能检索使用，尽管不是完美迁移。我想，那一天可能 5 年甚至 10 年之后才会到来。当然，对想让自己的笔记持续形式完美的人来说，现在开始就要做好“一生抉择”。

**Reply by Zuo-Peng** [reply]
笔记这个事情真是一直变化的，技术在变，时代在变。每一款笔记软件都会经历自己的一个爆发期，然后平静，然后瓶颈，然后可能就摆烂了，然后就是要么卖了，要么不行了。所以我个人认为很难有一款笔记软件能够长青，都是有寿命的，寿命取决于用户的数量和粘性。就拿最原始的word来说，其用户数量和粘性非常之大，所以说，用word记笔记，几乎不用担心未来的数据丢失等问题哈哈哈。
当然word不太一样啦。回到笔记软件，notion这类依靠database，自家格式比较封闭的软件，其实要非常重视数据迁移的问题，虽然notion，甚至像ob的base功能，都是各家软件支持各自，没有统一的标准，那在数据迁移时必然会遇到格式错乱的问题。
如果想像楼主说的，让自己的笔记形式持续完美的话，确实要选择好标准统一的格式，例如说.md文件，外面套个样式，不会有任何问题，直接使用vscode来记笔记，非常原始，但是非常稳定。或者说latex也是一样。

所以我个人折腾了那么久，基本上要给自己打一个预防针，几乎选择任何一款笔记软件，都有未来数据迁移的麻烦，未来是发展的，万一有更好的软件出来，又心动了，又开始迁移了，这都不好说。因此，告诉自己，数据迁移和管理是个人工作的一部分，是很正常且必要的一部分，就不必为了所选的笔记软件而焦虑或担忧了。

**Reply by Pathsis** [reply]
@Zuo-Peng 感觉 Obsidian 的寿命应该会很长。倒不必非得用 Word 或 vscode。
